src/Npgsql/NpgsqlConnection.cs

```diff
@@ -201,7 +201,22 @@ void OpenInternal()
             // have Password populated and should not be overwritten.
             if (Password == null)
             {
-                Password = Settings.Password;
+                if (Settings.Password != null)
+                {
+                    Password = Settings.Password;
+                }
+                else if (Settings.Password == null && !IntegratedSecurity)
```

Comment:
Also, no need to check `!IntegratedSecurity`, it's OK for the password to be set even if IntegratedSecurity is on. This continues the previous behavior where the Password field is populated regardless of IntegratedSecurity.

Although it may look weird, remember that it is PostgreSQL which determines what kind of authentication is going to take place, depending on how pg_hba.conf is set up. Providing both a password and IntegratedSecurity allows users to say "if the server wants a password use that, if it wants SSPI/GSS use that".